server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -954,6 +961,11 @@ public Builder creationDate(long creationDate) {
             return this;
         }
 
+        public Builder lifecycleOriginationDate(long originationDate) {
```

Comment:
I don't think we need the equivalent to this in x-pack, `IndexMetaData` has settings that are considered "core" to an index (we don't have builders for every one of them)